### PR TITLE
feat: add organizations as readonly access

### LINKED ISSUE #82: Allow the whole AWS Org for readonly
## Describe the Feature

In scenarios where there are many AWS accounts in the AWS Org, it is often more practical to allow the whole org
to pull images than dozens individual accounts.

## Expected Behavior

From each AWS account users and services can execute non-modifying actions, e.g. pulling the images, list images, etc.

## Describe Ideal Solution

A pseudo-policy:

```
      {
        "Sid": "AllowOrgPull",
        "Effect": "Allow",
        "Principal": "*",
        "Condition": {
          "StringEquals": {
            "aws:PrincipalOrgID": ${jsonencode(data.aws_organizations_organization.current.id)}
          }
        },
        "Action": [
          "ecr:BatchGetImage",
          "ecr:BatchCheckLayerAvailability",
          "ecr:CreateRepository",
          "ecr:DescribeImages",
          "ecr:DescribeImageScanFindings",
          "ecr:DescribeRepositories",
          "ecr:GetAuthorizationToken",
          "ecr:GetDownloadUrlForLayer",
          "ecr:GetLifecyclePolicy",
          "ecr:GetLifecyclePolicyPreview",
          "ecr:GetRepositoryPolicy",
          "ecr:ListImages",
          "ecr:ListTagsForResource"
        ]
      },
```

### PR DESCRIPTION
**Pull request by dragosmc**
## what
* Add the ability to have organizations as trustees (read-only) for the ECR repository

## why
* As described in #82, it's sometimes useful to allow an entire organization to consume images from a centralized repository

## references
* closes #82

### CODE REVIEW

**Comment by dragosmc** [comment]
Although functionally the PR is ready I want to update the readme too. Is this done automatically by the pipeline? Sorry, I couldn't find instructions on how to do it automatically.

File: main.tf [hunk 1]
```diff
@@ -131,6 +156,27 @@ data "aws_iam_policy_document" "resource_full_access" {
 
     actions = ["ecr:*"]
   }
+
+  dynamic "statement" {
```
Comment (nitrocode) [inline main.tf:160]: why include this block in the full access document? shouldn't it only be in the read only document?

**Comment by daleric-xyz** [comment]
Any update on this? @dragosmc

**Comment by dragosmc** [comment]
Apologies, didn't get a chance to revisit and address the comments. I expect to have an updated PR over the weekend.

**Comment by dragosmc** [comment]
I believe this is ready. Unfortunately, I cannot run the tests locally due to an issue with bats on MacOS

**Comment by jtyr** [comment]
The feature introduced by this PR is very useful. @dragosmc, please could you rebase the PR?

**Comment by Gowiem** [comment]
@dragosmc friendly ping on rebasing this as it does seem like people want this. If you can get this updated and conflicts resolved, then I'll review and we'll get this merged. Feel free to ping me when it's ready. Thanks!

**Comment by dragosmc** [comment]
Hi @Gowiem - I'll get this rebased and changed over the weekend. 👍

**Comment by dragosmc** [comment]
I renamed my fork's branch and this PR was closed automatically. A new one is open here https://github.com/cloudposse/terraform-aws-ecr/pull/106

**Pull request closed without merging**